src/test/shared/activationReloadState.test.ts

```diff
@@ -6,112 +6,115 @@
 'use strict'
 
 import * as assert from 'assert'
-import * as vscode from 'vscode'
 import {
     ACTIVATION_LAUNCH_PATH_KEY,
     ActivationReloadState,
     SAM_INIT_RUNTIME_KEY,
     SAM_INIT_IMAGE_BOOLEAN_KEY,
 } from '../../shared/activationReloadState'
-import { FakeExtensionContext } from '../fakeExtensionContext'
+import { ext } from '../../shared/extensionGlobals'
 
-class TestActivationReloadState extends ActivationReloadState {
-    public constructor(private readonly context: vscode.ExtensionContext) {
-        super()
-    }
+describe('ActivationReloadState', async function () {
+    const activationReloadState = new ActivationReloadState()
 
-    protected get extensionContext(): vscode.ExtensionContext {
-        return this.context
-    }
-}
+    beforeEach(function () {
+        activationReloadState.clearSamInitState()
+    })
+
+    afterEach(function () {
+        activationReloadState.clearSamInitState()
+    })
 
-describe('ActivationReloadState', async function() {
-    let extensionContext: FakeExtensionContext
-    let activationReloadState: ActivationReloadState
+    it('decides ext.didReload()', async function () {
+        await ext.context.globalState.update(ACTIVATION_LAUNCH_PATH_KEY, undefined)
```

Comment:
Probably minor considering how ephemeral the configurations here are, but this could potentially cause issues on developer's machines while running tests; either existing configurations cause issues, or this can affect a user's existing globalState.

Mostly moot for this scenario, but just be aware if we look to expand this sort of thing.

Reply:
We use FakeContext in the globalSetup.ts, so it should be fine.